### DISCUSSION TITLE
Server Status, Memory and CPU Load: When should i start to worry about?

### DISCUSSION THREAD

**Discussion by unudelaradio** [AzuraCast/AzuraCast#6982] [Q&A] [answered]
Hello!

I'm not very familiar with Server Status details in Azuracast Admin. When should i start to worry about my resources? At what point? How can i use the info displayed in Memory and CPU Load to avoid overloads, and any issues with my stations.

I attached a screenshot with my current Server Status. I have 4 radio stations configured in my Azuracast instance.
- 1 station is offline
- 3 stations are running with AutoDJ and 2 of them have Stereo Tool enabled

Is my server status ok right now? There are any overbusy risks? When should i stop adding stations?

![screenshot](https://github.com/AzuraCast/AzuraCast/assets/74244061/c599897e-3318-454d-a35f-fa8f1c642876)

**Comment by gAlleb** [comment, marked as answer]
Well, with such output you should probably never worry)) Everything is great and can handle more stations if you want) You have no CPU steal or wait. Enough RAM and some free RAM too. 
You can start worrying when `Load Average` will start approaching `4.0`.  Your CPU load should ideally stay below 4.00 and not exceed it (1.0 for every CPU). When and if you'll be running out of RAM - you can add more or use swap.

**Reply by unudelaradio** [reply]
Oh, Thank God!  

I was a little bit worried about using Stereo Tool. This means i can also use it on the 3rd running station too. Now is clear i have enough resources. 

Should i worry if on another server that i have, i see quite often some _Steal_ and _Wait_ percentages, but the `Load Average` is still very good, under `2.0`? The server has the same configuration: 4 CPU and 4GB of RAM.

I will upload a screenshot from the other server. I'm using an old version of AzuraCast, with only 2 radio stations running, no Stereo Tool. 

Everything works fine on the server. But I'm particularly interesed in how this kind of situation may affect the stations and the system. Does that _Steal_ and _Wait_ percentages are some warnings? Something i should worry about?

![screedshot2](https://github.com/AzuraCast/AzuraCast/assets/74244061/100a4f7b-a129-46b5-9f32-863d6fe9c4c6)

**Reply by gAlleb** [reply]
Well, when steal is under 3-5% I would say all is good. It may vary during day/night. Steal time is the percentage of time a virtual CPU within a cloud server involuntarily waits on a physical CPU for its processing time.
Your wait is very low too.  If you see `steal` value increase - then you could ask your service provider for assistance because they can manipulate the load of the node where your server is. (Cause Azuracast won't increase your steal)

I would say this server is good too. Average load is very low. Nothing to worry about.

**Reply by unudelaradio** [reply]
Many thanks for the answers!

**Comment by wppronto** [comment]
You just always want some headroom. Don't worry if the memory looks fully utilized as long as a reasonably large portion shows up as "cached". That means the operating system is reserving that for future use.

**Comment by Heimatradio** [comment]
i'm running 8 stations, 6 with StereoTool 
![Unbenannt-2](https://github.com/AzuraCast/AzuraCast/assets/150570533/54e51fb8-d2ec-46f4-8c48-4bb094dadf2d)
